framework/SupervisedLearning/ARMA.py

```diff
@@ -58,23 +58,35 @@ def __init__(self,messageHandler,**kwargs):
                            and printing messages
       @ In, kwargs: an arbitrary dictionary of keywords and values
     """
+    # general infrastructure
     supervisedLearning.__init__(self,messageHandler,**kwargs)
     self.printTag          = 'ARMA'
     self._dynamicHandling  = True # This ROM is able to manage the time-series on its own.
+    # training storage
     self.trainingData      = {} # holds normalized ('norm') and original ('raw') training data, by target
     self.cdfParams         = {} # dictionary of fitted CDF parameters, by target
-    self.fourierResults    = {} # dictionary of Fourier results, by target
     self.armaResult        = {} # dictionary of assorted useful arma information, by target
     self.correlations      = [] # list of correlated variables
+    self.fourierResults    = {} # dictionary of Fourier results, by target
+    # training parameters
+    self.fourierParams     = {} # dict of Fourier training params, by target (if requested, otherwise not present)
     self.Pmax              = kwargs.get('Pmax', 3) # bounds for autoregressive lag
     self.Pmin              = kwargs.get('Pmin', 0)
     self.Qmax              = kwargs.get('Qmax', 3) # bounds for moving average lag
     self.Qmin              = kwargs.get('Qmin', 0)
+    self.segments          = kwargs.get('segments', 1)
+    # data manipulation
     self.reseedCopies      = kwargs.get('reseedCopies',True)
-    self.outTruncation     = kwargs.get('outTruncation', None) # Additional parameters to allow user to specify the time series to be all positive or all negative
+    self.outTruncation = {'positive':set(),'negative':set()} # store truncation requests
     self.pivotParameterID  = kwargs['pivotParameter']
     self.pivotParameterValues = None  # In here we store the values of the pivot parameter (e.g. Time)
     self.seed              = kwargs.get('seed',None)
+    self.zeroFilterTarget  = kwargs.get('ZeroFilter',None) # SOLAR HACK
+    self.zeroFilterMask    = None # mask of places where zftarget is zero, or None if unused
```

Comment:
zftarget -> zeroFilterTarget

Reply:
We do not regulate the contents of comments.  If such a standard is adopted, or if the contents of the comment is unclear, I'd be happy to revise it.